src/common/factory.ts

```diff
@@ -54,4 +82,111 @@ export abstract class TurboBaseFactory {
       paymentService,
     });
   }
+
+  protected abstract getSigner({
+    providedPrivateKey,
+    providedSigner,
+    providedWalletAdapter,
+    logger,
+    token,
+  }: {
+    providedSigner: TurboSigner | undefined;
+    providedPrivateKey: TurboWallet | undefined;
+    providedWalletAdapter: WalletAdapter | undefined;
+    token: TokenType;
+    logger: TurboWinstonLogger;
+  }): TurboDataItemAbstractSigner;
+
+  protected abstract getAuthenticatedUploadService(
+    config: TurboAuthenticatedUploadServiceConfiguration,
+  ): TurboAuthenticatedUploadServiceInterface;
+
+  protected getAuthenticatedTurbo({
+    privateKey,
+    signer: providedSigner,
+    paymentServiceConfig = {},
+    uploadServiceConfig = {},
+    token,
+    gatewayUrl,
+    tokenMap,
+    tokenTools,
+    logger,
+    walletAdapter,
+  }: TurboAuthenticatedConfiguration & { logger: TurboWinstonLogger }) {
+    token = token === 'pol' ? 'matic' : token;
+
+    if (!token) {
```

Comment:
this `common` level factory de-duplicates the code from the `node` and `web` factories. we defer to the child classes for importing node and web specific uploaders and signers via abstract methods to keep the imports clean